content/articles/implement-a-collapsible-view-using-a-cardview/index.md

```diff
@@ -0,0 +1,768 @@
+               tools:visibility="gone" />
+
+
+       </androidx.constraintlayout.widget.ConstraintLayout>
+
+   </com.google.android.material.card.MaterialCardView>
+
+
+</androidx.constraintlayout.widget.ConstraintLayout>
+```
+
+You are all set. Now time for some java.
+
+
+#### Step nine: Add OnClick Listener to the ImageView
+In this step, we will set an `onClick` listeners to the show `imageView`.
+ 
+In your `Main Activity.java` file, add the following lines of code:
+
+```java
+public class MainActivity extends AppCompatActivity {
+   CardView cardView;
+   ImageView arrow;
+   Group hiddenGroup;
+
+   @Override
+   protected void onCreate(Bundle savedInstanceState) {
+       super.onCreate(savedInstanceState);
+       setContentView(R.layout.activity_main);
+       cardView = findViewById(R.id.base_cardview);
+       arrow = findViewById(R.id.show);
+       hiddenGroup = findViewById(R.id.card_group);
+
+       arrow.setOnClickListener(view -> {
+           if(hiddenGroup.getVisibility() == View.VISIBLE){
+               TransitionManager.beginDelayedTransition(cardView, new AutoTransition());
+               hiddenGroup.setVisibility(View.GONE);
+               arrow.setImageResource(android.R.drawable.arrow_down_float);
+           }
+           else {
+               TransitionManager.beginDelayedTransition(cardView, new AutoTransition());
+               hiddenGroup.setVisibility(View.VISIBLE);
+               arrow.setImageResource(android.R.drawable.arrow_up_float);
+           }
+       });
+
+   }
+}
+```
+
+Let us debunk this code:
+
+First, we referenced the views performing actions. These views are the `base_cardView`, `show` ImageView, and the `card_group`. Remember that you do not need to reference all 12 views anymore. This is because you have grouped them into the `card_group`.
+
+Next, set an `onClickListener` on the `show` ImageView. When this image is clicked, you would want to either show or hide the `card_group`.
+ 
+Inside the `onClickListerner`, pass the `CardView` as the root view to run the transition, which would make the collapsing and expanding of the card a lot nicer and give it an animation effect
+ 
+Finally, check if the group’s visibility is visible. If it is, you set its visibility to gone and change the image resource of the ImageView to `arrow_down_float`. Which kind of looks like a show-more icon.
+ 
+Else, if the group is not visible, then you would want to show it. Thus change the visibility of the group to Visible. Set the image resource of the ImageView to `arrow_up_float`, which looks like a show-less icon.
+ 
+Nice work!
+ 
+Let us run our app. Our collapsable card view should look this way:
+
+[![Collapsing Card View]({engineering-education/implement-a-collapsible-view-using-a-cardview/expanded_card.png})]({/engineering-education/implement-a-collapsible-view-using-a-cardview/collapsable_cardview.mp4} "Collapsing Card View")
+
+Nice right?
+  
+### Conclusion
```

Comment:
Also, I prefer if you have a downloadable APK of the app. You can host the APK on Google drive and share the link here.